src/Illuminate/Database/Connectors/ConnectionFactory.php

```diff
@@ -198,33 +187,31 @@ public function createConnector(array $config)
     /**
      * Create a new connection instance.
      *
-     * @param  string   $driver
-     * @param  \PDO     $connection
-     * @param  string   $database
-     * @param  string   $prefix
-     * @param  array    $config
+     * @param  string             $driver
+     * @param  ConnectorInterface $connector
+     * @param  string             $database
+     * @param  string             $prefix
+     * @param  array              $config
      * @return \Illuminate\Database\Connection
-     *
-     * @throws \InvalidArgumentException
      */
-    protected function createConnection($driver, PDO $connection, $database, $prefix = '', array $config = [])
+    protected function createConnection($driver, ConnectorInterface $connector, $database, $prefix = '', array $config = [])
     {
         if ($this->container->bound($key = "db.connection.{$driver}")) {
-            return $this->container->make($key, [$connection, $database, $prefix, $config]);
+            return $this->container->make($key, [$connector, $database, $prefix, $config]);
```

Comment:
This could potentially be a huge breaking BC for people extending/creating custom connection. E.g: oracle etc.